### ISSUE TITLE
[java] AvoidUsingOctalValues triggers on non-octal double literals with a leading 0

### ISSUE THREAD

**Issue by Gold856** [pmd/pmd#5007]
**Affects PMD Version:** 7.1.0

**Rule:** [AvoidUsingOctalValues](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#avoidusingoctalvalues)

**Description:**

When a double literal has a leading 0, and ends with either `d` or `.0`, the literal is not treated as an octal value. However, PMD still reports a violation.

**Code Sample demonstrating the issue:**

```java
double literal1 = 017.0; // Not octal!
double literal2 = 017d; // Also not octal!
double octal = 017; // Definitely octal!
```

**Expected outcome:**

PMD should only report a violation at line 3 only, but it reports on all three.

**Running PMD through:** *Gradle*
